tests/e2e/default/targets/target-aggregates.wdio-spec.js

```diff
@@ -103,134 +83,96 @@ const updateSettings = async (targetsConfig, user, contactSummary) => {
 
       await commonElements.goToAnalytics();
       await analytics.goToTargetAggregates(true);
-      expect(await element.all(by.css('#target-aggregates-list ul li')).count()).toEqual(0);
+      expect((await analytics.aggregateList()).length).to.equal(0);
       expect(
-        await element(by.css('.content-pane .item-content.empty-selection.selection-error')).isDisplayed()
-      ).toEqual(true);
+        await (await analytics.emptySelectionError()).isDisplayed()
+      ).to.be.true;
     });
   });
 
   describe('as a user with a home place', () => {
-    const password = 'passwordSUP3RS3CR37!';
-
-    const parentPlace = {
-      _id: 'PARENT_PLACE',
-      type: 'district_hospital',
-      name: 'Big Parent Hospital'
-    };
-
-    const otherParentPlace = {
-      _id: 'OTHER_PLACE',
-      type: 'district_hospital',
-      name: 'Smaller Hospital'
-    };
-
-    const user = {
-      username: 'supervisor',
-      password: password,
-      place: parentPlace._id,
-      contact: {
-        _id: 'fixture:user:supervisor',
-        name: 'Supervisor'
-      },
-      roles: ['national_admin'],
-      known: true,
-      language: 'en',
-    };
-
-    const names = [ 'Clarissa', 'Prometheus', 'Alabama', 'Jasmine', 'Danielle' ];
-
+    const parentPlace = placeFactory.place().build({ type: 'district_hospital' });
+    const otherParentPlace = placeFactory.place().build({ type: 'district_hospital' });
+    const user = userFactory.build({ place: parentPlace._id, roles: ['program_officer'] });
+    const names = ['Clarissa', 'Prometheus', 'Alabama', 'Jasmine', 'Danielle'];
     const genPlace = (parent, idx = false) => {
-      const place = {
-        _id: uuid(),
-        type: 'health_center',
-        name: randomString(8),
-        parent: { _id: parent._id },
-        reported_date: moment().valueOf(),
-      };
-
-      const contact = {
-        _id: uuid(),
-        type: 'person',
-        name:  idx === false ? randomString(8) : names[idx],
-        parent: { _id: place._id, parent: place.parent },
-        reported_date: moment().valueOf(),
-      };
-
+      const place = placeFactory.place().build({ type: 'health_center', parent: { _id: parent._id } });
+      const contact = personFactory.build({
+        name: idx === false ? randomString(8) : names[idx],
+        parent: { _id: place._id, parent: place.parent }
+      });
       place.contact = { _id: contact._id, parent: contact.parent };
-
       return [place, contact];
     };
-
     const docs = _.flattenDeep([
       Array.from({ length: 5 }).map((e, i) => genPlace(parentPlace, i)),
       Array.from({ length: 5 }).map(() => genPlace(otherParentPlace)),
     ]);
-
     const genTitle = (title) => ({ en: title });
-
     const docTags = [
       // current targets
       moment().format('YYYY-MM'),
       // next month targets, in case the reporting period switches mid-test
       moment().date(1).add(1, 'month').format('YYYY-MM'),
```

Comment:
if in real life it's the end of the month or the beginning of the month in the system, will the targets be miss calculated and fail the test? We had some bad experiences with setting dates for targets before.

Reply:
@lorerod did you check this one? This can trigger flaky tests at certain days of the month when running tests.

Reply:
@latin-panda sorry I missed this comment. Do you suggest creating the targets without a reporting_period?